Warning when building

When trying release the 1.2 version, I noticed the following lines:

```
gcc -DHAVE_CONFIG_H -I.  -g -O3 -Wall -I/usr/X11R6/include  -I/usr/include -I/usr/include -I. -DPREFIX=\""/usr"\" -I/usr/include/giblib -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/PKGS/scrot/scrot-1.2=. -fstack-protector-strong -Wformat -Werror=format-security -c -o note.o note.c
In file included from /usr/include/string.h:494,
                 from /usr/include/X11/Xos.h:62,
                 from scrot.h:36,
                 from main.c:33:
In function ‘strncat’,
    inlined from ‘im_printf’ at main.c:704:7:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:136:10: warning: ‘__builtin___strncat_chk’ output may be truncated copying 1 byte from a string of length 4095 [-Wstringop-truncation]
  136 |   return __builtin___strncat_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Thanks!

--- a/src/main.c
+++ b/src/main.c
@@ -700,8 +700,11 @@ im_printf(char *str, struct tm *tm,
           strncat(ret, c, 1);
           break;
       }
-    } else
-      strncat(ret, c, 1);
+    } else {
+      const size_t len = strlen(ret);
+      ret[len] = *c;
+      ret[len + 1] = '\0';
+    }
   }
   return gib_estrdup(ret);
 }

--- a/src/main.c
+++ b/src/main.c
@@ -623,7 +623,11 @@ im_printf(char *str, struct tm *tm,
   struct stat st;
 
   ret[0] = '\0';
-  strftime(strf, 4095, str, tm);
+
+  if (strftime(strf, 4095, str, tm) == 0) {
+    fprintf(stderr, "strftime returned 0\n");
+    exit(EXIT_FAILURE);
+  }
 
   for (c = strf; *c != '\0'; c++) {
     if (*c == '$') {

